python/tests/test_parsimony.py

```diff
@@ -635,56 +741,81 @@ class TestParsimonyMissingData(TestParsimonyBase):
     Tests that we correctly map_mutations when we have missing data.
     """
 
-    def test_all_missing(self):
-        for n in range(2, 10):
-            ts = msprime.simulate(n, random_seed=2)
-            tree = ts.first()
+    @pytest.mark.parametrize("n", range(2, 10))
+    def test_all_missing(self, n):
+        ts = msprime.simulate(n, random_seed=2)
```

Comment:
The places you have `ts = msprime.simulate(n, random_seed=2)` could use a session-scoped parameterised fixture. https://docs.pytest.org/en/stable/fixture.html#fixture-parametrize

Reply:
might pick that up later - it's hard to know where to draw the line when updating the test code!